### PR TITLE
CORE-8760 Prepare notary for finality flow integration

### PR DESCRIPTION
**Pull request by nkovacsx**
### Overview

This PR is doing some small fixes that will help us with the integration with the finality flow.
These fixes has been tested in an integrated environment with @vlajos.
Fixes:
- Create interface for the plugin factory
- Fix cluster setup for `UtxoLedgerTests`
- Fix plugin name in smoke tests
- Use `MemberLookup` in the worker selector service
- Add `signatureSpec` to the signature metadata
- Add `UsedByPersistence` to factory

### Testing

Tested with integrated `UtxoFinalityFlow` from https://github.com/corda/corda-runtime-os/pull/2729

### CODE REVIEW

File: components/virtual-node/sandbox-notary-impl/src/main/kotlin/net/corda/ledger/notary/plugin/factory/impl/PluggableNotaryClientFlowFactoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        installProviders(
+            context.getMetadataServices()
+        )
+    }
+
+    @Suspendable
+    private fun installProviders(providers: Set<PluggableNotaryClientFlowProvider>) {
+        providers.forEach { provider ->
+
+            logger.info("Installing plugin provider class: ${provider.javaClass.name}")
```
Comment (lankydan) [inline components/virtual-node/sandbox-notary-impl/src/main/kotlin/net/corda/ledger/notary/plugin/factory/impl/PluggableNotaryClientFlowFactoryImpl.kt]: I think debug might be better for this log.

Comment (nkovacsx) [reply components/virtual-node/sandbox-notary-impl/src/main/kotlin/net/corda/ledger/notary/plugin/factory/impl/PluggableNotaryClientFlowFactoryImpl.kt]: Good point, downgraded to debug

File: components/virtual-node/sandbox-notary-impl/build.gradle [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+dependencies {
+    compileOnly 'org.osgi:osgi.annotation'
+    compileOnly 'org.osgi:org.osgi.service.component.annotations'
+
+    api 'net.corda:corda-notary-plugin'
+    implementation project(':components:ledger:notary-worker-selection')
+    implementation project(':libs:sandbox')
+    implementation project(':libs:virtual-node:sandbox-group-context')
+    implementation project(':components:virtual-node:sandbox-notary')
```
Comment (relyafi) [inline components/virtual-node/sandbox-notary-impl/build.gradle]:
```suggestion
    implementation project(':components:ledger:notary-worker-selection')
    implementation project(':components:virtual-node:sandbox-notary')
    implementation project(':libs:sandbox')
    implementation project(':libs:virtual-node:sandbox-group-context')
```
Ordering.

**Review by relyafi — DISMISSED** [review]
One trivial comment, maybe not worth addressing in isolation.

Otherwise LGTM - good to see this coming together!

**Review by relyafi — APPROVED** [review]

**Review by wzur-r3 — APPROVED** [review]
Gradle files LGTM

**Pull request merged**